keps/sig-api-machinery/1929-built-in-default/README.md

```diff
@@ -271,17 +271,13 @@ example further down).
 
 ### Default publishing
 
-CRDs do not publish default via the OpenAPI spec on the /openapi/v2
-endpoint today, but we have to decide what to do with the new default
-values of built-in types. The kube-openapi generated schemas will
-include them.
-
-We proppose to:
-1. leave CRDs not publishing defaults
-2. filter out defaults in the built-in type schemas too in a first step
-3. but possibly reconsider this behaviour in the future when the
-necessity of defaults for client-side merging (e.g. through kustomize)
-and influences on the ecosystem of default mis-use is better understood.
+Before 1.22, CRDs and build-in types were not exposing their default values through the `/openapi/v2` endpoint but will expose them in Kubernetes v1.22.
+
+These default values can be valuable for generating documentation, and can be required for clients that are trying to merge associative lists with defaulted keys.
+
+Please note that clients absolutely MUST NOT use these default values to populate fields that haven't been otherwise specified, as noted in the OpenAPI documentation:
+> The default value is the one that the server uses if the client does not supply the parameter value in the request.
+
```

Comment:
```suggestion

Both for built-in types and CRDs, the defaults exposed in the OpenAPI are not showing the full picture. There's always the possibility for defaults to happen implicitly. For built-in, values can be defaulted by defaulting functions which would not expose the values in the schema. Both built-in and CRDs can have mutating webhooks that will default values who will not be exposed in the schema either. Except for associative lists with defaulted keys, these defaults should be treated as informational only and are best-effort.
```

Reply:
should we use `x-kubernetes-default-doc-only` instead to make this clear?

Reply:
Depends what you mean by "doc-only". If you mean "informational", then sure, but that's already the purpose of the "default" field.

Reply:
= best effort: don't depend on its availability.